mitiq/zne/scaling/folding.py

```diff
@@ -297,12 +293,11 @@ def fold_global(
 
     if num_to_fold > 0:
         folded += Circuit(
-            [inverse(operations[-num_to_fold:])], [operations[-num_to_fold:]]
+            [inverse(base_circuit[-num_to_fold:])],
+            [base_circuit[-num_to_fold:]],
```

Comment:
This is currently inserting gates based on moments. Want to access operations in the moment and add them in a similar manner. For this reason, the test `test_global_fold_stretch_factor_eight_terminal_measurements` and the output circuit is not the intended globally folded circuit.